BigBaseV2/src/hooks/protections/received_clone_sync.cpp

```diff
@@ -25,38 +29,77 @@ namespace big {
-
-		if (!net_obj) return 2;
-
-		if (!sync_tree || sync_type < 0 || sync_type > 14) invalidsync = true;
-
-		if (net_obj->m_object_type != sync_type) invalidsync = true;
-
-		//TO BE ADDED
-		//Node specific entity type checks
-
-
-		if (invalidsync) {
-			if (g->notifications.invalid_sync.log)
-				LOG(WARNING) << "Invalid sync: " << "Type: " << sync_type << " Tree name: " << tree_name << " From: " << src->get_name();
-			if (g->notifications.invalid_sync.notify)
-				g_notification_service->push_warning(fmt::format("Invalid Sync from {}", src->get_name()), fmt::format("Type {} in sync tree {}", sync_type, tree_name));
-
-			return 2;
+		if (auto sync_tree = g_pointers->m_get_sync_tree_for_type(mgr, sync_type); sync_tree && NETWORK::NETWORK_IS_SESSION_ACTIVE())
+		{
+			if (auto net_obj = g_pointers->m_get_net_object(mgr, obj_id, true); net_obj)
+			{
+				auto tree_name = g_pointers->m_get_sync_type_info(sync_type, 0);
+				bool invalidsync = false;
+
+				if (sync_type < 0 || sync_type > 14)
+				{
+					if (g->notifications.out_of_allowed_range_sync_type.log)
+						LOG(WARNING) << "Out of range sync: " << "Type: " << sync_type << " Tree name: " << tree_name << " From: " << src->get_name();
+					if (g->notifications.out_of_allowed_range_sync_type.notify)
+						g_notification_service->push_warning(fmt::format("Out Of Allowed Sync Range from {}", src->get_name()), fmt::format("Type {} in sync tree {}", sync_type, tree_name));
+				}
+				else if (net_obj->m_object_type != sync_type)
+				{
+					if (g->notifications.mismatch_sync_type.log)
+						LOG(WARNING) << "Mismatch sync: " << "Type: " << sync_type << " Tree name: " << tree_name << " From: " << src->get_name();
+					if (g->notifications.mismatch_sync_type.notify)
+						g_notification_service->push_warning(fmt::format("Mismatch Sync from {}", src->get_name()), fmt::format("Type {} in sync tree {}", sync_type, tree_name));
+
+					return 2;
+				}
+				else
+				{
+					if (auto game_obj = net_obj->GetGameObject(); game_obj)
+					{
+						if (auto model_info = game_obj->m_model_info)
+						{
+							if (!STREAMING::IS_MODEL_VALID(model_info->m_model_hash))
+							{
+								return 2;
+							}
+							//This hurts me, but Yim, go make a fucking CVehicleModelInfo class. Please.
+							if (*reinterpret_cast<uint64_t*>(reinterpret_cast<uint64_t>(model_info) + 0x340) != *reinterpret_cast<uint64_t*>(reinterpret_cast<uint64_t>(get_model_data(model_info->m_model_hash)) + 0x340))
+							{
+								return 2;
+							}
+							if (model_info->m_model_type != get_model_data(model_info->m_model_hash)->m_model_type)
+							{
+								return 2;
+							}
+							//Since we can access model hashes, we can also block the sync of stuff like SLOD. I can't be asked to implement this further, so I'll leave this comment for those who want to add it.
+						}
+					}
+				}
+			}
+			else
+			{
+				if (sync_tree != 6) //We don't want to not sync a player, so we ensure it's not a ped
```

Comment:
`else if` exists, no need for separating the else and the if.

Also isn't the `sync_tree` variable a pointer?

Reply:
Oops